sentry-core/src/main/java/io/sentry/core/protocol/App.java

```diff
@@ -77,9 +80,29 @@ public void setAppBuild(String appBuild) {
     this.appBuild = appBuild;
   }
 
+  @TestOnly
+  Map<String, Object> getUnknown() {
+    return unknown;
+  }
+
   @ApiStatus.Internal
   @Override
   public void acceptUnknownProperties(Map<String, Object> unknown) {
     this.unknown = unknown;
   }
+
+  /**
+   * Clones an App aka deep copy
+   *
+   * @return the cloned App
+   * @throws CloneNotSupportedException if object is not cloneable
+   */
+  @Override
+  public @NotNull App clone() throws CloneNotSupportedException {
+    final App clone = (App) super.clone();
```

Comment:
Is this cloning via reflection?
Should we implementing the assignments by hand instead?

Reply:
It executes shallow clone using native JVM methods - there's no reflection overhead.

Reply:
That being said I have never used `clone` method before and rather went with copy constructors for similar use cases.